### DISCUSSION TITLE
How to use Field Sampling with a Field on different grid than U,V ?

### DISCUSSION THREAD

**Discussion by VincentIMEDEA** [Parcels-code/Parcels#1719] [Q&A] [answered]
## Question
Hi,
I followed the tutorial on **Field Sampling**.
Everything works great if the Field is on the same grid as U and V.

Is it possible to have a Field on a **different grid** (and even with different time values)?
(I could do the spatial/temporal interpolation offline first, but maybe there is a better way)

Let s say that U and V read their "longitude" and "latitude" from the "uvfile" netcdf file while the Field SSS read its "longitude_sss" and "latitude_sss" from the "sssfile" netcdf file.
(same with time and time_sss; time_sss encompasses time)

## Supporting code/error messages

```python

filenames = {
    "U": {"lon": uvfile, "lat": uvfile, "data": uvfile},
    "V": {"lon": uvfile, "lat": uvfile, "data": uvfile},
    "SSS": {"lon": sssfile, "lat": sssfile, "data": sssfile},
}

variables = {'U': 'u',
            'V': 'v',
            'SSS': 'sss'}

dimensions = {'U': {'time':'time',
                    'lon': 'longitude', 
                    'lat': 'latitude'},
            'V': {'time':'time',
                    'lon': 'longitude', 
                    'lat': 'latitude'},
            'SSS': {'time':'time_sss',
                    'lon': 'longitude_sss', 
                    'lat': 'latitude_sss'}}

fieldset = FieldSet.from_netcdf(filenames, variables, dimensions,
                                allow_time_extrapolation=False, transpose=False, mesh='spherical')
```

```text
# Paste your error message within this block
```

**Comment by michaeldenes** [comment, marked as answer]
Hi @VincentIMEDEA, a simple answer to your question is: "Yes, absolutely!"

The `FieldSet` object has a nice functions `add_field()` that allows you to add different fields with different grids and temporal resolutions very easily.

An example of this is in the [`plasticparcels`](https://plastic.oceanparcels.org/en/latest/index.html) project, where we combine 1/12th degree daily ocean model output with 6 hourly wind/wave data (on a different grid), and weekly averaged biogeochemical model data, all in one fieldset.

Typically, you won't call `from_netcdf()` (or other associated methods) just once, rather, incrementally build your fieldset. Starting with your hydrodynamic data, you can create `fieldset`. Then you can create your SSS fieldset, say `fieldset_sss`, and you can add this to the `fieldset` via `fieldset.add_field(fieldset_sss.sss)` for example.

Happy to answer any other questions! Cheers :)

**Reply by michaeldenes** [reply]
To add, re-reading your code, it may be possible to do it that way too, I've never tried!

**Reply by VincentIMEDEA** [reply]
thanks !!
it works like a charm.
As you suggested, I just created a separate fields for SSS and use the fieldset.add_field function:
```py
filenames_sss = {"SSS": {"lon": mesh_mask_sss, "lat": mesh_mask_sss, "data": sssfiles}}
variables_sss = {'SSS': 'sss'}
dimensions_sss = {'SSS': {'time':'time',
                    'lon': 'longitude', 
                    'lat': 'latitude'}}
fieldset_sss = FieldSet.from_netcdf(filenames_sss, variables_sss, dimensions_sss,
                                allow_time_extrapolation=False, transpose=False, mesh='spherical')

fieldset.add_field(fieldset_sss.SSS)
```
